### PR TITLE
Support SCTP port mapping

### PR DESCRIPTION
**Pull request by ishidawataru**
Signed-off-by: Wataru Ishida <ishida.wataru@lab.ntt.co.jp>

**- What I did**

This PR add support for SCTP port mapping.

Please see https://github.com/moby/moby/pull/33922 for the details.

**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by aaronlehmann** [comment]
I see the PR in go-connections is https://github.com/docker/go-connections/pull/41

Let's merge that first, then we can update this PR to vendor the actual go-connections repository instead of a fork.

**Comment by ishidawataru** [comment]
@aaronlehmann Sure. I'll update this PR after https://github.com/docker/go-connections/pull/41 get merged.

However, I think https://github.com/docker/libnetwork/pull/1825 is the first PR we need to merge for  SCTP port mapping feature.
Could you also take a look on that PR?

**Comment by AkihiroSuda** [comment]
rebased

**Comment by dnephin** [comment]
Ok, but let's not merge this until the moby/moby PR is merged.

**Comment by thaJeztah** [comment]
Perhaps we want to update the tests for https://github.com/docker/cli/pull/581 to include SCTP

**Comment by thaJeztah** [comment]
https://github.com/moby/moby/pull/33922 was merged

File: opts/port.go [hunk 1]
```diff
@@ -49,7 +49,8 @@ func (p *PortOpt) Set(value string) error {
 
 			switch key {
 			case portOptProtocol:
-				if value != string(swarm.PortConfigProtocolTCP) && value != string(swarm.PortConfigProtocolUDP) {
+				// TODO: use swarm.PortConfigProtocolSCTP (vendor moby/moby#33922)
```
Comment (cpuguy83) [inline opts/port.go]: Looks like we just need to vendor this in now?

**Comment by thaJeztah** [comment]
Opened a vendor PR for moby/moby: https://github.com/docker/cli/pull/892

**Comment by AkihiroSuda** [comment]
Will rebase soon after #892 gets merged, thanks

**Comment by AkihiroSuda** [comment]
done

**Comment by thaJeztah** [comment]
Thanks @AkihiroSuda - this does need a follow-up to update the documentation, for example, the [`docker service create`](https://github.com/docker/cli/blob/ce985d04cedd089ac4bbf49cb267385c23ad7a90/docs/reference/commandline/service_create.md#publish-service-ports-externally-to-the-swarm--p---publish) reference doc mentions the accepted protocols, but there may be other locations (opened https://github.com/docker/docker.github.io/issues/6039 for tracking that)

**Pull request merged**